src/modules/storage/LedgerStorage.ts

```diff
@@ -849,4 +849,137 @@ export class LedgerStorage extends Storages
                 AND tx_outputs.used != 1;`;
         return this.query(sql, [address]);
     }
+
+    /**
+     * Provides a history of transactions.
+     * Returns data sorted in descending order by block height.
+     * The most recent transaction is located at the front.
+     *
+     * @param addresses An array of addresses that want to be inquired.
+     * @param page_size Maximum record count that can be obtained from one query
+     * @param page      The number on the page, this value begins with 1
+     */
+    public getTransactionHistory (addresses: string[], page_size: number, page: number) : Promise<any[]>
+    {
+        let accounts = addresses.map((n) => `'${n}'`).join(',');
+
+        let sql =
+            `SELECT
+                TX.address,
+                TX.block_height as height,
+                TX.tx_hash,
+                TX.type,
+                TX.unlock_height,
+                (SUM(TX.income) - SUM(TX.spend)) as amount
+            FROM
+            (
+                    SELECT
+                        S.address,
+                        T.block_height,
+                        T.tx_hash,
+                        T.tx_index,
+                        T.type,
+                        T.unlock_height,
+                        0 as income,
+                        SUM(S.amount) as spend
+                    FROM
+                        transactions AS T,
+                        tx_inputs AS I,
+                        tx_outputs AS S
+                    WHERE
+                        T.block_height = I.block_height
+                        AND T.tx_index = I.tx_index
+                        AND I.utxo = S.utxo_key
+                        AND S.address in (${accounts})
```

Comment:
You must also join the blocks table.
It's not on the block table, but there may only be transactions, tx_input, tx_ouputs

Reply:
Is it the standard JOIN method for sqlite?
Shouldn't we do this?
FROM transactions T INNER JOIN tx_inputs I ON (T.block_height = I.block_height)

Reply:
> You must also join the blocks table.
> It's not on the block table, but there may only be transactions, tx_input, tx_ouputs

When is that the case?

Reply:
> Is it the standard JOIN method for sqlite?
> Shouldn't we do this?
> FROM transactions T INNER JOIN tx_inputs I ON (T.block_height = I.block_height)

That's a good idea.

Reply:
For data integrity.

Reply:
Okay, That's a good idea